src/main/java/org/infernalstudios/foodeffects/FoodEffectsEvents.java

```diff
@@ -34,109 +31,56 @@
-                () -> COMMON.COOKIE_EFFECT_AMPLIFIER.get()
-            )
-        );
-        EFFECT_MAP.put(
-            Items.PUMPKIN_PIE,
-            new EffectData(
-                () -> MOB_EFFECTS.getValue(new ResourceLocation(COMMON.PUMPKIN_PIE_EFFECT.get())),
-                () -> (int)(COMMON.PUMPKIN_PIE_EFFECT_DURATION.get() * 20),
-                () -> COMMON.PUMPKIN_PIE_EFFECT_AMPLIFIER.get()
-            )
-        );
-        EFFECT_MAP.put(
-            Items.HONEY_BOTTLE,
-            new EffectData(
-                () -> MOB_EFFECTS.getValue(new ResourceLocation(COMMON.HONEY_BOTTLE_EFFECT.get())),
-                () -> (int)(COMMON.HONEY_BOTTLE_EFFECT_DURATION.get() * 20),
-                () -> COMMON.HONEY_BOTTLE_EFFECT_AMPLIFIER.get()
-            )
-        );
-        EFFECT_MAP.put(
-            Items.BAKED_POTATO,
-            new EffectData(
-                () -> MOB_EFFECTS.getValue(new ResourceLocation(COMMON.BAKED_POTATO_EFFECT.get())),
-                () -> (int)(COMMON.BAKED_POTATO_EFFECT_DURATION.get() * 20),
-                () -> COMMON.BAKED_POTATO_EFFECT_AMPLIFIER.get()
-            )
-        );
-    }
-
+    protected static final Map<Item, List<EffectData>> EFFECT_MAP = new ConcurrentHashMap<>();
 
     @SubscribeEvent
     public void onLivingEntityUseItemFinish(LivingEntityUseItemEvent.Finish event) {
         if (event.getEntity() instanceof Player player) {
-            ItemStack stack = event.getItem();
-    
-            if (stack.getItem() == Items.DRIED_KELP) {
-                player.removeEffect(MobEffects.POISON);
-                player.removeEffect(MobEffects.CONFUSION);
-                player.removeEffect(MobEffects.BLINDNESS);
+            onEat(player, event.getItem().getItem());
+        }
+    }
+
+    public static void onEat(Player player, Item item) {
+        if (item == Items.DRIED_KELP) {
+            player.removeEffect(MobEffects.POISON);
+            player.removeEffect(MobEffects.CONFUSION);
+            player.removeEffect(MobEffects.BLINDNESS);
+        }
+        List<EffectData> effectData = EFFECT_MAP.get(item);
+        if (effectData == null) {
+            return;
+        }
+        for (EffectData effect : effectData) {
+            if (effect.getDuration() == 0) {
+                player.removeEffect(effect.getEffect());
             } else {
-                EffectData effectData = EFFECT_MAP.get(stack.getItem());
-                if (effectData != null) {
-                    MobEffectInstance effect = new MobEffectInstance(
-                        effectData.effect.get(),
-                        effectData.duration.get(),
-                        effectData.amplifier.get()
-                    );
-                    player.addEffect(effect);
-                }
+                player.addEffect(
+                    new MobEffectInstance(
+                        effect.getEffect(),
+                        effect.getDuration(),
```

Comment:
Sorry, missed this the first time! Previously the config duration was in seconds and needed to be multiplied by 20 to convert to ticks. Has this been changed?

Reply:
It's calculated on (de)serialization. Defaults already have them in ticks.